rfd/0133-connect-my-computer.md

```diff
@@ -0,0 +1,621 @@
+---
+authors: Grzegorz Zdunek (grzegorz.zdunek@goteleport.com), Rafał Cieślak (rafal.cieslak@goteleport.com)
+state: draft
+---
+
+# RFD 133 - Connect My Computer
+
+## Required Approvers
+
+* Engineering: @zmb3
+* Product: (@xinding33 || @klizhentas )
+* Security: @jentfoo
+
+## What
+
+Connect My Computer (further abbreviated as CMC) enables the user of Teleport Connect to add the
+device running Teleport Connect to the cluster as an SSH node.
+
+The MVP achieves that by downloading the teleport binary, creating the relevant role and the join
+token and then starting the agent from within the Electron app. Since the device is added as a
+regular node, this poses a security concern of enabling any cluster user with sufficient permissions
+to log into the device running Connect My Computer. This and other concerns are addressed in [the
+Security section](#security).
+
+The goal of this RFD is to gather feedback on the core of Connect My Computer, that is managing the
+node agent running on the same device as Connect. Decisions on how to drive the users toward this
+feature, such as adding a Discover tile or deep linking from the Web UI to Connect, are not covered
+in this RFD and will be covered in future RFDs.
+
+The wireframes included in the RFD are supposed to convey the general idea behind the UI. The setup
+tab and the status tab are likely to stay in the same format. The top bar UI and the CTAs are likely
+to change, although they are likely to receive no extra new features. If you have feedback on the UI
+which goes beyond the general flow and towards the specifics, please leave it directly on [the
+document in Figma](https://www.figma.com/file/jeKtOZrFsIVOtKH7yd8Xiy/Connect-My-Computer-UI?type=design).
+
+## Why
+
+The idea behind Connect My Computer is to reduce [the time to
+magic](https://beneinstein.medium.com/min-time-to-magic-c938d9ea3473). By enabling the user to add a
+new node to the cluster with just a few clicks, we hope to get them to the wow moment as quickly as
+possible. We want to show that configuring and accessing the infrastructure with Teleport is fast
+and easy. This will be further amplified by adding new features further down the line, such as
+sharing sessions.
+
+## Details
+
+After signing into a Team plan cluster, which currently has no connected nodes, the user sees a
```

Comment:
Sorry for reviving this thread but I think it'll actually be rare for a user to download Teleport Connect and log in to a completely empty cluster. I believe the ultimate goal of CMC to be virality. The feature is meant to bring more users to Teleport. To that end, I think we should show the "Connect your Computer" button as long as the current user's computer is NOT connected. Of course, we don't have to make it as visible as when there are no resources connect at all.

Reply:
That's more or less the plan with regards to the UI. See the new designs made by Kenny, [the empty state](https://www.figma.com/file/jeKtOZrFsIVOtKH7yd8Xiy/Connect-My-Computer-UI?type=design&node-id=118%3A7026&mode=design&t=4U4ao1wmn4JDsXLj-1) and [the button to manage Connect My Computer](https://www.figma.com/file/jeKtOZrFsIVOtKH7yd8Xiy/Connect-My-Computer-UI?type=design&node-id=118%3A7030&mode=design&t=4U4ao1wmn4JDsXLj-1) in the top right of the resource table.

The button will always be visible, as long as the user's account meets all the requirements to run Connect My Computer. The copy in the empty state will be shown only if there's no resources at all.